src/groth16/prover.rs

```diff
@@ -452,38 +571,56 @@ where
             ))
         })
         .collect::<Result<Vec<_>, SynthesisError>>()?;
+
+    drop(a_inputs_source);
+    drop(a_aux_source);
+    drop(b_g1_inputs_source);
+    drop(b_g1_aux_source);
+    drop(b_g2_inputs_source);
+    drop(b_g2_aux_source);
     drop(multiexp_kern);
 
-    let proofs = h_s
-        .into_iter()
-        .zip(l_s.into_iter())
-        .zip(inputs.into_iter())
-        .zip(r_s.into_iter())
-        .zip(s_s.into_iter())
+    info!("proofs prep");
+    let gs = r_s
+        .par_iter()
+        .zip(s_s.par_iter())
```

Comment:
Looks the same as above (i.e. `par_iter` should be in a threadpool/install scope) if I'm understanding this right.  There's another `par_iter` above that I didn't comment on as well.

Reply:
fixed